root: add v6.36.00

wdconinc opened the pull request:
This PR adds to `root` the newest version, [6.36.00](https://github.com/root-project/root/compare/v6-34-08...v6-36-00). Since this is a bigger release than the bugfix releases in #50642, it's submitted separately. Full release notes at https://root.cern/doc/v636/release-notes.html.

Notable changes:
- `cxxstd=23` support
- `asimage` now has companions `asimage_tiff`, `builtin_gif`, `builtin_jpeg`, `builtin_png`

wdconinc committed:
root: add v6.36.00

wdconinc committed:
root: pkg modifications for 6.36.00

wdconinc committed:
root: fix style

wdconinc commented on var/spack/repos/spack_repo/builtin/packages/root/package.py:
```diff
@@ -238,12 +239,14 @@ class Root(CMakePackage):
         default=False,
         description="Enable support for TMultilayerPerceptron " "classes' federation",
```
```suggestion
        description="Enable support for TMultilayerPerceptron classes' federation",
```

wdconinc committed:
root: reflow string

wdconinc committed:
root: turn self-referential mold dependency into conflict

alalazo reviewed (changes requested):
It seems we have a few legitimate build failures?

wdconinc committed:
root: depends_on giflib when @6.36: (previously vendored)

wdconinc commented:
> It seems we have a few legitimate build failures?

Added missing `giflib` (not using builtin anymore), and rebased to get past conflict due to #50642.

greenc-FNAL commented:
@wdconinc @tgamblin @marcpaterno Note: #50600 predates this, includes 6.36.00-rc1, and was awaiting only the formal release.
